### DISCUSSION TITLE
pm2 on Reboot not working for me

### DISCUSSION THREAD

**Discussion by jreising326** [tagyoureit/nodejs-poolController#781] [Q&A]
Hi Folks, I setup my Pi with a different user name so I'm running njsPC out of /home/joe.  I think in doing this pm2 isn't finding the path to njsPC, Panel, and REM.  Logorate is running and is given a PID, the others show an EOENT error in the logs, no PID assigned.  I tried adding my home directory to the command that set PATH$, but it gave me an error.  I think if I can put my home directory into the PATH$ before pm2 runs on Bootup it might solve this problem.  Anyone else run into this?  I think I'd prefer to use pm2 over other methods if I can resolve this issue.

**Comment by rstrouse** [comment]
When you type pm2 startup it will give you a command something like the following depending on the installation location and such.
```
sudo env PATH=$PATH:/home/pi/.config/nvm/versions/node/v18.16.0/bin home/pi/.config/nvm/versions/node/v18.16.0/lib/node_modules/pm2/bin/pm2 startup systemd -u pi --hp /home/pi
```

Which you need to run to add it to your path and the systemd setup.  This is in step 7 of the wiki.

**Reply by jreising326** [reply]
Yes I did run that command, even changed script to reflect /home/joe.  Just went through the Wiki once more, same result.  Here's the error message that has me thinking it's a Path issue, but I could be wrong.  I might just create a Pi user and copy over the files and see what happens.
2023-06-07T17:02:59: PM2 error: Error: spawn node ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

**Reply by rstrouse** [reply]
That is weird.  It can also be a permissions issue.  Does home/joe/.config/nvm/versions/node/v18.16.0 exist?

**Reply by jreising326** [reply]
yes that directory exists.  At this point I might just start from scratch.  Tried creating a pi account, copying files over and running as that user and now the servers aren't even running from either account.  Oops.  Not sure this has anything to do with anything but I installed the 64-bit version of Raspian.

**Reply by jreising326** [reply]
Fresh install 32 Bit OS, default pi username, working fine now.  Not sure if there is a bug in pm2 or if I screwed up/corrupted something.

**Reply by rstrouse** [reply]
Yeah the 64-bit raspian is a questionable.  It has been in beta for a very long time and never seems to move out of that state.  There are libraries that just don't work yet and getting a remote desktop to connect was not even possible last time I tried.
